Mathlib/Algebra/Module/Equiv/Basic.lean

```diff
@@ -495,17 +495,28 @@ section Semiring
 
 open LinearMap
 
-variable {M₂₁ M₂₂ : Type*} [Semiring R] [AddCommMonoid M₁] [AddCommMonoid M₂]
-  [AddCommMonoid M₂₁] [AddCommMonoid M₂₂] [Module R M₁] [Module R M₂] [Module R M₂₁] [Module R M₂₂]
+section Semilinear
+
+variable {R₁ R₂ R₁' R₂' : Type*} {M₁ M₂ M₁' M₂' : Type*}
```

Comment:
Maybe an ascii-art diagram like the one at line 566 would be useful here as well?

Reply:
That sounds like a good solution.